addons/test_mail/tests/test_mail_composer.py

```diff
@@ -0,0 +1,288 @@
+        composer = self.env['mail.compose.message'].with_context({
+            'default_composition_mode': 'comment',
+            'default_model': self.test_record._name,
+            'default_res_id': self.test_record.id,
+        }).create({
+            'subject': 'My amazing subject',
+            'body': '<p>Test Body</p>',
+        })
+
+        # creation values are taken
+        self.assertEqual(composer.subject, 'My amazing subject')
+        self.assertEqual(composer.body, '<p>Test Body</p>')
+        self.assertEqual(composer.mail_server_id.id, False)
+
+        # changing template should update its content
+        composer.write({'template_id': self.template.id})
+        # currently onchange necessary
+        composer.onchange_template_id_wrapper()
+        self.assertEqual(composer.subject, 'About %s' % self.test_record.name)
+        self.assertEqual(composer.body, '<p>Hello %s</p>' % self.test_record.name)
+        self.assertEqual(composer.mail_server_id, self.template.mail_server_id)
+
+        # manual values is kept over template
+        composer.write({'subject': 'Back to my amazing subject'})
+        self.assertEqual(composer.subject, 'Back to my amazing subject')
+
+    @users('employee')
+    def test_mail_composer_content_comment_wtpl(self):
+        composer = self.env['mail.compose.message'].with_context({
+            'default_composition_mode': 'comment',
+            'default_model': self.test_record._name,
+            'default_res_id': self.test_record.id,
+        }).create({
+            'template_id': self.template.id,
+        })
+
+        # creation values from from template
+        composer.write({'template_id': self.template.id})
+        # currently onchange necessary
+        composer.onchange_template_id_wrapper()
+        self.assertEqual(composer.subject, 'About %s' % self.test_record.name)
+        self.assertEqual(composer.body, '<p>Hello %s</p>' % self.test_record.name)
+        self.assertEqual(composer.mail_server_id, self.template.mail_server_id)
+
+    @users('employee')
+    def test_mail_composer_content_mass(self):
+        composer = self.env['mail.compose.message'].with_context({
+            'default_composition_mode': 'mass_mail',
+            'default_model': self.test_record._name,
+        }).create({
+            'subject': 'My amazing subject',
+            'body': '<p>Test Body</p>',
+            'mail_server_id': False,
+            'template_id': self.template.id,
+        })
+
+        # creation values are taken
+        self.assertEqual(composer.subject, 'My amazing subject')
+        self.assertEqual(composer.body, '<p>Test Body</p>')
+        self.assertEqual(composer.mail_server_id.id, False)
+
+        # erasing template should erase its content
+        composer.write({'template_id': False})
+        # currently onchange necessary
+        composer.onchange_template_id_wrapper()
+        self.assertEqual(composer.subject, 'My amazing subject')
```

Comment:
It doesn't erase the subject?

Reply:
Some strange behavior indeed... I am asserting current behavior anyway.